core/logger/manager.php

```diff
@@ -50,6 +52,40 @@ public function shutdown() {
 		$this->get_logger()->log( $item );
 	}
 
+	public function rest_error_handler( $error_number, $error_message, $error_file, $error_line ) {
+		$error = new \WP_Error( $error_number, $error_message, [
+			'type' => $error_number,
+			'message' => $error_message,
+			'file' => $error_file,
+			'line' => $error_line,
+		] );
+
+		// Notify $e.data.
+		if ( ! headers_sent() ) {
+			header( 'Content-Type: application/json; charset=UTF-8' );
+
+			http_response_code( 500 );
+
+			if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
+				echo wp_json_encode( $error->get_error_data() );
+			} else {
+				echo wp_json_encode( [
+					'message' => 'Server error, see Elementor => System Info',
+				] );
+			}
+		}
+
+		$this->shutdown( $error->get_error_data() );
+
+		exit;
+	}
+
+	public function register_rest_error_handler() {
+		if ( ! \Elementor\Data\Manager::instance()->is_internal() ) {
```

Comment:
The data manger should register and use this manger.

Reply:
Handled: https://github.com/elementor/elementor/pull/11853/commits/14aee2dd7f9f201c63f393c14f54395f47e04a21